lib/valueflow.cpp

```diff
@@ -3623,11 +3623,17 @@ static void valueFlowLifetime(TokenList *tokenlist, SymbolDatabase*, ErrorLogger
             if (!Token::Match(parent, ". %name% ("))
                 continue;
 
+            bool isContainerOfPointers = true;
+            const Token* containerTypeToken = tok->valueType()->containerTypeToken;
+            if (containerTypeToken && Token::simpleMatch(containerTypeToken->previous(), "<") && containerTypeToken->previous()->link()) {
```

Comment:
this assumes that the container only has 1 template parameter. I feel 99% ok with that. For instance `std::array` has more template parameters but I am not sure if `valueFlowLifetime` is interesting for `std::array` elements. well.. what do you think is this code ok?

Reply:
Well `std::vector` can take two parameters as well, so this should probably be updated. If we change the example to this:

```cpp
struct A {
    ~A() { std::cout << "deleted\n"; }
};

int main() {
    std::vector<A *, std::allocator<A*>> v;
    A *a = new A();
    v.push_back(a);
    A *b = v.back();
    v.pop_back();
    std::cout << "pop_back\n";
    delete b;
}
```

It will fail. I should probably use `parseDecl` instead.